test/libsolidity/SMTCheckerTest.cpp

```diff
@@ -82,6 +82,9 @@ SMTCheckerTest::SMTCheckerTest(string const& _filename): SyntaxTest(_filename, E
 	if (m_modelCheckerSettings.solvers.none() || m_modelCheckerSettings.engine.none())
 		m_shouldRun = false;
 
+	// Disable the default timeout to force resource limit.
+	m_modelCheckerSettings.timeout = {};
```

Comment:
Oh, so setting this to a `std::nullopt` is essentially equivalent as having it set to `0`, i.e. no timeout? Still not sure why this has to be `optional`.

Reply:
No, there's 3 cases here:
- not nullopt and not 0: timeout is used and has a nonzero value
- not nullopt and 0: timeout is used and equals infinity
- nullopt: timeout is not used at all. In this case, the solvers use a `resource limit` which is based on internal operations and not on time.